### PR TITLE
add util.promisify

### PR DESCRIPTION
**Pull request by legendecas**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

File: src/js/util.js [hunk 1]
```diff
@@ -250,6 +250,33 @@ function deprecate(fn, msg, code) {
...
+  }
+  return function promisified() {
+    var args = Array.prototype.slice.call(arguments, 0);
+
+    return new Promise((resolve, reject) => {
+      args.push(callback);
+      original.apply(this, args);
+
+      function callback(err, result) {
+        if (err != null) {
```
Comment (yorkie) [inline src/js/util.js]: Should we verify this is an `Error` instance?

File: src/js/util.js [hunk 2]
```diff
@@ -250,6 +250,33 @@ function deprecate(fn, msg, code) {
   return deprecated;
 }
 
+var customPromisifySymbol = 'util:promisify:custom';
```
Comment (yorkie) [inline src/js/util.js]: This looks like the symbol, so i think it's better to flag a TODO or FIXME as the following:

```js
// FIXME(legendecas): change to use formal `Symbol` when it is ready.
```

File: src/js/util.js (hunk 1 above)
Comment (legendecas) [reply src/js/util.js]: No

Comment (yorkie) [reply src/js/util.js]: I saw the same function on Node.js looks like this:

```js
if (err)
  return reject(err)
```

Why do we make this difference here?

Comment (legendecas) [reply src/js/util.js]: `if (err)` would be true if err is truthy, `err != null` can only be true if err is `null` or `undefined`.

Comment (yorkie) [reply src/js/util.js]: Yay! How about fixing it in Node.js core, either?

**Comment by yorkie** [comment]
@legendecas Looks good to me, please add document as well :)

**Pull request merged**